### PR TITLE
test(database, e2e): remove workaround for emulator rules

### PR DESCRIPTION
**Pull request by davidepalazzo**
### Description

It seems that the latest version of the Emulator loads the database rules correctly, which means we can get rid of the workarounds.

The `e2e` tests run successfully after updating the `helpers.js` file.

### Related issues
https://github.com/invertase/react-native-firebase/issues/5876

### Release Summary
Remove workarounds in e2e for database Emulator

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [X] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [X] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [X] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

🔥

### CODE REVIEW

File: tests/ios/Podfile.lock [hunk 1]
```diff
@@ -1168,72 +1168,73 @@ PODS:
     - React-jsi (= 0.67.3)
     - React-logger (= 0.67.3)
     - React-perflogger (= 0.67.3)
-  - RNFBAnalytics (15.2.0):
+  - RNFBAnalytics (15.3.0):
```
Comment (mikehardy) [inline tests/ios/Podfile.lock:1171]: Don't commit the lock file as part of an update, I consider dependency management (including lock file updates) a separate task and like to keep it in it's own commits on it's own PRs (see #6482 and the commit series there)

Just dropping this file from the commit entirely is the way to go

File: packages/database/e2e/helpers.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
-const testingUtils = require('@firebase/rules-unit-testing');
+// const testingUtils = require('@firebase/rules-unit-testing');
```
Comment (mikehardy) [inline packages/database/e2e/helpers.js:1]: Source control allows us to be bold - don't leave commented out code, if you've got something working, delete the lines

I love diffs that remove more lines than they add ;-)

**Review by mikehardy — CHANGES_REQUESTED** [review]
This looks great! Thanks for testing it, I haven't had time myself yet.
It looks like the CLA was signed by a variation on your name, you may  need to alter the commit email address via some git config work, or add an email address to your github account then re-try ?
I left a couple comments on the PR itself, but they are minor in the grand scheme: the real work was checking it, which you've done, thank you

File: packages/database/e2e/helpers.js (hunk 2 above)
Comment (davidepalazzo) [reply packages/database/e2e/helpers.js:1]: Sorry, this slipped through the cracks while I was rebasing to edit the commits with the right author for the CLA.

**Comment by davidepalazzo** [comment]
Thanks for your speedy response @mikehardy. I addressed everything you mentioned in your comments and sorted out the CLA with my correct signed account (I had it done in the past so not sure what happen there).

In my local the e2e runs smoothly:
Screen Shot 2022-08-28 at 9 25 39 am


Let's see if the pipeline likes that too!

**Review by mikehardy — DISMISSED** [review]
Fantastic - this looks perfect

**Comment by mikehardy** [comment]
Oh my! Just saw this one was still sitting around, apologies. Still getting back up to speed since summer travels.
I just squashed the three commits to one, rebased it against current main, and re-pushed it so CI will hopefully clear
Then we can get this in, it's a nice cleanup

**Review by mikehardy — APPROVED** [review]
still looks good to me, assuming CI goes green of course :-)

**Comment by mikehardy** [comment]
Hmm - not sure what the explanation is, but currently failing every time on iOS at least (android is having unrelated issues) in CI with

```

  database().ref().on()
    ✔ throws if event type is invalid
    ✔ throws if callback is not a function
    ✔ throws if cancel callback is not a function
    ✔ throws if context is not an object
    ✔ should callback with an initial value
    - should callback multiple times when the value changes
16:56:14.777 detox[17936] WARN:  at e2e/init.js:41:15 
 ⚠️ A test failed:
16:56:14.778 detox[17936] WARN:  at e2e/init.js:42:15 
 ️   ->  should cancel when something goes wrong
16:56:14.779 detox[17936] WARN:  at e2e/init.js:49:13 
 ️   ->  Retrying in 5 seconds ... (1)
16:56:24.799 detox[17936] WARN:  at e2e/init.js:46:15 
    🔴  Retry #1 failed...
16:56:24.800 detox[17936] WARN:  at e2e/init.js:49:13 
 ️   ->  Retrying in 10 seconds ... (2)
16:56:39.803 detox[17936] WARN:  at e2e/init.js:46:15 
    🔴  Retry #2 failed...
16:56:39.804 detox[17936] WARN:  at e2e/init.js:49:13 
 ️   ->  Retrying in 15 seconds ... (3)
16:56:59.808 detox[17936] WARN:  at e2e/init.js:46:15 
    🔴  Retry #3 failed...
16:56:59.809 detox[17936] WARN:  at e2e/init.js:49:13 
 ️   ->  Retrying in 20 seconds ... (4)
    1) should cancel when something goes wrong


  319 passing (6m)
  23 pending
  1 failing

  1) database().ref().on()
       should cancel when something goes wrong:
     Error: Spy was not called within timeout period.
      at Timeout._onTimeout (node_modules/@react-native-firebase/private-tests-helpers/lib/index.js:3:824)
      at listOnTimeout (node:internal/timers:559:17)
      at processTimers (node:internal/timers:502:7)
```

**Comment by davidepalazzo** [comment]
No worries at all! Between travelling and work I only had time to have a look at this now.

I retested the changes I made with a fresh clone from `main` and I'm getting the same failing test in my local as the CI. It looks like we still need to keep the workaround for now after all. ☹️

**Comment by mikehardy** [comment]
Well, at least we are not crazy since we have reproducible results. Or with only a sample size of 2 perhaps it is still sunspots or something and we're just falling prey to confirmation bias. But I'll trust the result - I guess we still need the workaround

It was a wonderful thought though, I do love purging workarounds - thanks a bunch for the attempt. Cheers
